File: src/code/java/MissingNumber.java
package code.java;

import java.util.Arrays;

public class MissingNumber {
    public int missingNumber(int[] nums) {
        return nums.length * (nums.length + 1) / 2 - Arrays.stream(nums).sum();
    }

    public static void main(String[] args) {
        var sol = new MissingNumber();
        System.out.println(sol.missingNumber(new int[]{3, 0, 1}));
        System.out.println(sol.missingNumber(new int[]{0, 1}));
        System.out.println(sol.missingNumber(new int[]{9, 6, 4, 2, 3, 5, 7, 0, 1}));
    }
}

File: src/code/dart/missing_number.dart
class Solution {
  int missingNumber(List<int> nums) =>
      nums.length * (nums.length + 1) ~/ 2 -
      nums.reduce((value, element) => value + element);
}

void main() {
  var sol = Solution();
  print(sol.missingNumber([3, 0, 1]));
  print(sol.missingNumber([0, 1]));
  print(sol.missingNumber([9, 6, 4, 2, 3, 5, 7, 0, 1]));
}
